be/src/vec/functions/function_timestamp.cpp

```diff
@@ -1182,6 +1184,139 @@ class FunctionOtherTypesToDateType : public IFunction {
     }
 };
 
+struct FromIso8601DateV2 {
+    static constexpr auto name = "from_iso8601_date";
+
+    static bool is_variadic() { return true; } //todo : check is or not is_variadic
+
+    static DataTypes get_variadic_argument_types() { return {std::make_shared<DataTypeString>()}; }
+
+    static DataTypePtr get_return_type_impl(const DataTypes& arguments) {
+        return make_nullable(std::make_shared<DataTypeDateV2>());
+    }
+
+    static Status execute(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                          size_t result, size_t input_rows_count) {
+        const auto* src_column_ptr = block.get_by_position(arguments[0]).column.get();
+
+        std::regex calendar_regex(R"((\d{4})(-)?(\d{2})(-)?(\d{2}))"); // YYYY-MM-DD or YYYYMMDD
+        std::regex year_month_regex(R"((\d{4})-(\d{2}))");             // YYYY-MM
+        std::regex year_only_regex(R"((\d{4}))");                      // YYYY
+        std::regex week_regex(
+                R"((\d{4})(-)?W(\d{2})(-)?(\d)?)"); // YYYY-Www or YYYY-Www-D or YYYYWww or YYYYWwwD
+        std::regex day_of_year_regex(R"((\d{4})(-)?(\d{3}))"); // YYYY-DDD or YYYYDDD
```

Comment:
Consider add `const` for them and move outside of `execute()` to improve the efficiency.